src/micro.c

```diff
@@ -53,10 +55,18 @@ micro_AddService(microService **new_m, natsConnection *nc, microServiceConfig *c
 
     MICRO_CALL(err, _clone_service_config(&m->cfg, cfg));
 
+    // Add the service to the connection.
+    MICRO_CALL(err, _attach_service_to_connection(m->nc, m));
+    MICRO_CALL(err, (m->refs++, NULL));
```

Comment:
I am not saying we don't need a ref count here, but it seems to me that the connection should be retained (inside `_attach_service_to_connection()` maybe?). Meaning that you probably want the connection object to be valid while a service exists, no?

Reply:
See the next thread